src/configuration.ts

```diff
@@ -58,15 +60,37 @@ export class RLSConfiguration {
         this.revealOutputChannelOn = RLSConfiguration.readRevealOutputChannelOn(configuration);
         this.updateOnStartup = configuration.get<boolean>('rust-client.updateOnStartup', true);
 
-        this.channel = configuration.get('rust-client.channel', 'nightly');
-        this.componentName = configuration.get('rust-client.rls-name', 'rls');
+        this.channel = configuration.get('rust-client.channel', 'default');
+        if (this.channel === 'default') {
+            try {
+                this.channel = RLSConfiguration.defaultChannel(this.rustupPath);
+            } catch (e) {
+                console.log(e);
+            }
+        }
```

Comment:
Could you move this code to rustup.rs?

Reply:
I thought about moving it there but most of the code in `rustup.ts` relies on the `RLSConfiguration` object being already initialised, and is async. I thought this code would be inconsistent with the rest of the code there.

Reply:
It looks like you're using `RLSConfiguration` here too though? I don't think that whether the code is sync or async should make too much difference as to where it should live.